### PR TITLE
fix: signing tx with restored account

### PR DESCRIPTION
**Pull request by anukulpandey**
- signing tx with imported accounts ( qr and json)
- export account with custom password 
- minor bug fixes

### CODE REVIEW

File: lib/utils/password_manager.dart [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    String oldPass = accountElems[1];
+    Map<String, dynamic> updatedAccountObj = {
+      'name': account.name,
+      'svg': account.svg,
+      'address': account.address,
+      'mnemonic': "${account.address}+$newPass"
+    };
+    final updatedAccount =
+        StoredAccount.fromString(jsonEncode(updatedAccountObj).toString());
+    ReefAppState.instance.storage.deleteAccount(account.address);
```
Comment (matjazonline) [inline lib/utils/password_manager.dart]: delete account through AccountCtrl

File: lib/utils/password_manager.dart [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    } catch (e) {
+      return false;
+    }
+  }
+
+  static void updateAccountPassword(
+      StoredAccount account, String newPass) async {
+    // to only split from the first occurence considering that password might contain '+' in it    
+    int delimiterIdx = account.mnemonic.indexOf("+");
+    List<String> accountElems = [
```
Comment (matjazonline) [inline lib/utils/password_manager.dart:23]: better to use split

Comment (matjazonline) [reply lib/utils/password_manager.dart:23]: this is used in several places so create a method

File: lib/utils/password_manager.dart [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  static void updateAccountPassword(
+      StoredAccount account, String newPass) async {
+    // to only split from the first occurence considering that password might contain '+' in it    
+    int delimiterIdx = account.mnemonic.indexOf("+");
```
Comment (matjazonline) [inline lib/utils/password_manager.dart:22]: crete a method to check if it's mnemonic or json import account

File: lib/utils/password_manager.dart [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    final accountPassChanged = await ReefAppState.instance.accountCtrl
+        .changeAccountPassword(account.address, newPass, oldPass);
+    print(
+        "CHANGING ACCOUNT PASSWORD FOR ${account.address} - CHANGED PASSWORD :  $accountPassChanged");
+  }
+
+  static void changePasswordForAll(String newPass) async {
+    final allAccounts = await ReefAppState.instance.storage.getAllAccounts();
+    for (StoredAccount account in allAccounts) {
+      if (account.mnemonic.contains("+")) {
```
Comment (matjazonline) [inline lib/utils/password_manager.dart]: method to check type of account recovery value type

File: lib/utils/password_manager.dart (hunk 2 above)
Comment (anukulpandey) [reply lib/utils/password_manager.dart:23]: - if we use split and the user has password which contains + , for example "anukul+pandey" is password then the array we will get will be [addr,anukul,pandey] , where as the password is anukul+pandey , so using delimiterIdx is better or else we will need to iterate the resulting array to concat password .

- yea will create a method for it

**Pull request merged**